### PR TITLE
Add tutorial based on TodoMVC example

### PR DESCRIPTION
**Pull request by hkasabas**

### CODE REVIEW

File: docs/tutorials/todomvc-gaudi.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
+# How "todo" Gaudi
+
+## Intro
+
+You've probably all heard of the famous "TodoMVC" application. It was originally written by Addy Osmani of Google, not as a specific application but more as a project that provides a variety of implementations of a simple to-do list application. It is used to show and compare different JavaScript frameworks, technologies and even performance benchmarks. It's gotten so famous that it even got it's own [website](https://todomvc.com/) and it's own [application specification](https://github.com/tastejs/todomvc/blob/master/app-spec.md).
```
Comment (mpranjic) [inline docs/tutorials/todomvc-gaudi.md]: I'd remove `all`

File: docs/tutorials/todomvc-gaudi.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+```js
+api {
+  entrypoint Todo {
+    // list all todos
+    list endpoint { order by { id desc } }
+    // create new todo
+    create endpoint {}
+    // update existing todo
+    update endpoint {}
+    // delete toso
```
Comment (mpranjic) [inline docs/tutorials/todomvc-gaudi.md]: toso -> todo

File: docs/tutorials/todomvc-gaudi.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+Since "TodoMVC" primarily defines a GUI application and as such implemented mostly as a GUI-only applications, to communicate with our Gaudi backend we could either customize one of the existing examples or write a completely new GUI for our tutorial. Examples available at [todomvc.com](https://todomvc.com/) were pretty outdated and unmaintained so we've decided to go with our own GUI implemented with [React](htps://react.dev) and [Typescript](https://www.typescriptlang.org/).
+
```
Comment (mpranjic) [inline docs/tutorials/todomvc-gaudi.md]: This whole introduction is rather long and probably not necessary, but I enjoyed reading it :) Should we keep it as is? How do you feel about it now that it's 2 weeks old? :P

We could shorten it and make it boring, eg:


This tutorial demonstrates how to write a simple application using Gaudi and React. We chose to implement a well-known UI app, TodoMVC, which is commonly used to demonstrate GUI frameworks (link to website). We have modified the app to utilize a Gaudi backend service as a data storage interface.

**Review by mpranjic — APPROVED** [review]

File: docs/tutorials/todomvc-gaudi.md (hunk 3 above)
Comment (hkasabas) [reply docs/tutorials/todomvc-gaudi.md]: I like stories so I'm biased. :-) I've refactored it a bit so it's a little bit "drier" than before but still has an intro story.

**Pull request merged**